subscription-service/src/main/kotlin/com/egm/stellio/subscription/service/SubscriptionService.kt

```diff
@@ -591,4 +586,42 @@ class SubscriptionService(
     private var rowToSub: (Row) -> String = { row ->
         row.get("sub", String::class.java)!!
     }
+
+    /**
+     * Function that returns all subscriptions that contain the 'timeInterval' attribute and
+     * when the 'lastNotification' has passed
+     *
+     * @return list of subscription
+     */
+    suspend fun getAllSubscriptions(): List<Subscription> {
+        val selectStatement =
+            """
+            SELECT subscription.id as sub_id, subscription.type as sub_type, name, created_at, modified_At, expires_at,
+                description, watched_attributes, time_interval, q,  notif_attributes, notif_format, endpoint_uri, 
+                endpoint_accept, endpoint_info,  status, times_sent, last_notification, last_failure, last_success, 
+                is_active, entity_info.id as entity_id, id_pattern, entity_info.type as entity_type, georel, geometry, 
+                coordinates, geoproperty
+            FROM subscription
+            LEFT JOIN entity_info ON entity_info.subscription_id = subscription.id
+            LEFT JOIN geometry_query ON geometry_query.subscription_id = subscription.id
+            WHERE time_interval IS NOT NULL
+            AND last_notification IS NULL 
+            OR ((EXTRACT(EPOCH FROM last_notification) + time_interval) < EXTRACT(EPOCH FROM :currentDate))
```

Comment:
- don't you miss parenthesis to group these 2 conditions?
- should not the condition between the 2 dates be in the opposite way?

Reply:
1. Yes I forgot a parenthesis
2. If the condition are in opposite way, It's going to return me notification when the TimeInterval are not past

Reply:
indeed for 2.